pkg/controller/core.oam.dev/v1alpha2/application/application_controller_test.go

```diff
@@ -331,6 +331,30 @@ spec:
 		Expect(envs[0].Value).Should(Equal("ccc"))
 	})
 
+	It("app step will set event", func() {
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "vela-test-app-without-trait-event",
+			},
+		}
+		appwithNoTrait.SetNamespace(ns.Name)
+		Expect(k8sClient.Create(ctx, ns)).Should(BeNil())
+		Expect(k8sClient.Create(ctx, appwithNoTrait.DeepCopyObject())).Should(BeNil())
+
+		appKey := client.ObjectKey{
+			Name:      appwithNoTrait.Name,
+			Namespace: appwithNoTrait.Namespace,
+		}
+		reconcileRetry(reconciler, reconcile.Request{NamespacedName: appKey})
+
+		events := collectEvents(recorder.Events)
+		Expect(len(events)).ShouldNot(Equal(0))
+		for _, event := range events {
+			Expect(event).Should(ContainSubstring(corev1.EventTypeNormal))
```

Comment:
Will other negtive case affect this test result? If I understand correct, this events will contain all events from reconcile in this suit test. 

We'd better compare more detailed events info, and narrow the scope.

Reply:
copy